src/coreclr/vm/appdomain.cpp

```diff
@@ -3778,39 +3769,40 @@ void AppDomain::AddUnmanagedImageToCache(LPCWSTR libraryName, NATIVE_LIBRARY_HAN
     CONTRACTL
     {
         THROWS;
-        GC_TRIGGERS;
+        GC_NOTRIGGER;
         MODE_ANY;
         PRECONDITION(CheckPointer(libraryName));
```

Comment:
Is `NULL` okay for the `NATIVE_LIBRARY_HANDLE`?

Reply:
Nope. Will add a precondition.